Dockerfile

```diff
@@ -1,4 +1,4 @@
-FROM ruby:3.3.0
+FROM ruby:3.3.1
```

Comment:
other references (like ruby-version) need to be updated too

\+ we should check Heroku supports the new version

Reply:
https://devcenter.heroku.com/articles/ruby-support#ruby-versions 3.3.1 supported

Reply:
https://devcenter.heroku.com/articles/ruby-support#libraries seems we can also upgrade our bundler to 2.5.6! fun